### PR TITLE
Add a (soft) deadline for requests for information

### LINKED ISSUE #7283: Add a deadline for requests for information
Part of #7172 

Reviewers can send information requests to developers, but the deadline isn't clear to either, so admins end up doing a lot of work clearing them up. Implementing a concrete deadline makes things clearer for both developers and reviewers:

* [x] Replace the "Is more info requested?" checkbox with a "Require developer to respond in 7 days" checkbox. Users with `Reviews:Admin` should be able to change the number of days.
* [x] When a developer uploads a new version for an add-on that has a pending info request, show it in the description step, with a checkbox indicating "This request for information has been resolved". The checkbox will clear the request and log a developer response indicating it.
* [x] Modify the request for information email to include the deadline.
* [x] Send another email to the developer a day before the deadline expires, explaining the add-on may be disabled if they don't respond.
* [x] Add button to "Clear information request" to the More Actions section (#7282), if the user has `Reviews:Admin` and the add-on has a pending info request. This should log an event in the review history.

### PR DESCRIPTION
**Pull request by diox**
Fix #7283 

* [x] Replace the "Is more info requested?" checkbox with a "Require developer to respond in 7 days" checkbox. Users with `Reviews:Admin` should be able to change the number of days.
* [x] When a developer uploads a new version for an add-on that has a pending info request, show it in the description step, with a checkbox indicating "This request for information has been resolved". The checkbox will clear the request and log a developer response indicating it.
* [x] Modify the request for information email to include the deadline.
* [x] Send another email to the developer a day before the deadline expires, explaining the add-on may be disabled if they don't respond.
* [x] Add button to "Clear information request" to the More Actions section (#7282), if the user has `Reviews:Admin` and the add-on has a pending info request. This should log an event in the review history.
* [x] We won't automatically remove expired flags, admins will follow up with them manually. So, there needs to be some view for admins to look at the expired ones

Note: although the code to send the reminder email is present, it's not yet sent by cron, I want QA to test it first. The command is meant to fire once per hour.

### CODE REVIEW

File: src/olympia/reviewers/models.py [hunk 1]
```diff
@@ -1197,19 +1209,3 @@ class Meta:
     def __unicode__(self):
         return u'[%s] private: |%s| public: |%s|' % (
             self.addon.name, self.private, self.public)
-
-
-@Whiteboard.on_change
-def watch_public_whiteboard_changes(old_attr=None, new_attr=None,
```
Comment (eviljeff) [inline src/olympia/reviewers/models.py]: why is this being removed?  I didn't see a mention of it.

Comment (diox) [reply src/olympia/reviewers/models.py]: The way I understood the issue, you need to use the new "Clear request for information" action, which is behind a `Reviews:Admin` check, to, well, clear a request for information now. So this bit looked obsolete.

**Comment by diox** [comment]
Yeah I think I'll tackle #7419 next - this PR was already big enough...

File: src/olympia/reviewers/models.py (hunk 1 above)
Comment (eviljeff) [reply src/olympia/reviewers/models.py]: Isn't this also called for developer changes to the Whiteboard though?  (The model is in `/reviewers/models.py` but it's used in devhub too, iirc)

Comment (diox) [reply src/olympia/reviewers/models.py]: Right, but developers now have an explicit checkbox when submitting/editing a version to clear the info request.

Comment (eviljeff) [reply src/olympia/reviewers/models.py]: Must have forgotten that part - it was a long patch!  Nothing to see here then.

**Comment by diox** [comment]
Just realized I forgot to log an event when an admin clears the info request. I'll add that now.

**Pull request merged**